rnd/autogpt_server/Dockerfile

```diff
@@ -24,27 +19,41 @@ ENV POETRY_VERSION=1.8.3 \
     PATH="$POETRY_HOME/bin:$PATH"
 RUN pip3 install poetry
 
-FROM server_base AS server_dependencies
+# Copy and install dependencies
+COPY rnd/autogpt_libs /app/rnd/autogpt_libs
+COPY rnd/autogpt_server/poetry.lock rnd/autogpt_server/pyproject.toml /app/rnd/autogpt_server/
+WORKDIR /app/rnd/autogpt_server
+RUN poetry config virtualenvs.create false \
+    && poetry install --no-interaction --no-ansi
+
+# Generate Prisma client
+COPY rnd/autogpt_server/schema.prisma ./
+RUN poetry config virtualenvs.create false && poetry run prisma generate
+
+FROM python:3.11-slim-buster AS server_dependencies
+
+WORKDIR /app
+
+# Copy only necessary files from builder
+COPY --from=builder /app /app
+COPY --from=builder /usr/local/lib/python3.11 /usr/local/lib/python3.11
+COPY --from=builder /usr/local/bin /usr/local/bin
```

Comment:
???

Reply:
This allows us to avoid needing to include all the build tools in the final container by copying across the built binaries and dependancies.

Reply:
What binaries are being built outside of the Poetry environment?

Reply:
YouTube-transcriber defo does, maybe others.